src/main/java/com/pusher/rest/PusherAbstract.java

```diff
@@ -257,11 +289,32 @@ public T trigger(final List<String> channels, final String eventName, final Obje
         Prerequisites.areValidChannels(channels);
         Prerequisites.isValidSocketId(socketId);
 
-        final String body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
+        final String eventBody;
```

Comment:
I see your point, but after reading the `isValidSocketId` logic, it seems that they intentionally didn't check if `socketId` is empty or null. They only do a regex check when it isn't null. (Which seems odd to me as well because I also expect to `socketId` to not be empty.

If you don't mind, let's not touch it and leave it as it was. I'll later check with other devs or other SDKs to see if they accept empty socketId or not.

Reply:
okay, agreed